processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,7 +86,12 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
-	return d.rb.Emit(), conventions.SchemaURL, nil
+	// The error is unhandled because we want to return successfully detected resources
+	// regardless of an error. The caller will properly handle any error hit while getting
+	// the cluster name.
+	clusterName, err := d.utils.getClusterName(ctx)
+	d.rb.SetK8sClusterName(clusterName)
```

Comment:
can you please wrap it in `if rb.config.K8sClusterName.Enabled {` so user don't run into errors issues if the atribute isn't needed

Reply:
Good call, will do.

Reply:
Actually, it looks like we don't have access to this since `config` is a private member of the `ResourceBuilder` struct in the `metadata` package. Is it worth re-architecting some of this to get access, or should we leave it as is?

Reply:
It should be easy to pass. Just add a new field to `detector` struct and set `cfg.ResourceAttributes` to it in `NewDetector` func